fe/lib/components/BrandSelect/BrandSelect.tsx

```diff
@@ -0,0 +1,119 @@
+import { ApolloClient, useQuery } from '@apollo/client';
+import {
+  FieldMessage,
+  Heading,
+  Inline,
+  Loader,
+  Stack,
+  Text,
+} from 'braid-design-system';
+import React, { useEffect, useState } from 'react';
+
+import {
+  AdvertisementBrandingFieldsFragment,
+  AdvertisementBrandingsQuery,
+  AdvertisementBrandingsQueryVariables,
+} from '../../types/seekApi.graphql';
+
+import { PaginatedBrands } from './PaginatedBrands';
+import { ADVERTISEMENT_BRANDINGS } from './queries';
+
+export interface BrandSelectProps {
+  client?: ApolloClient<unknown>;
+  hirerId: string;
+  errorMessage?: string;
+  initialBrandId?: string;
+  /**
+   * Type definition from the `Controller` component that's provided by `react-hook-form`.
+   */
+  onChange: (...event: any[]) => void;
+}
+
+export const BrandSelect = ({
+  client,
+  hirerId,
+  errorMessage,
+  onChange,
+  initialBrandId,
+}: BrandSelectProps) => {
+  const PAGE_SIZE = 4;
+
+  const [variables, setVariables] =
+    useState<AdvertisementBrandingsQueryVariables>({
+      hirerId,
+      first: PAGE_SIZE,
+    });
+
+  const { data, previousData, loading, error } = useQuery<
+    AdvertisementBrandingsQuery,
+    AdvertisementBrandingsQueryVariables
+  >(ADVERTISEMENT_BRANDINGS, {
+    ...(client && { client }),
+    variables,
+  });
+
+  const [selectedBrandId, setSelectedBrandId] = useState(initialBrandId);
+
+  useEffect(() => {
```

Comment:
useEffect here to take `initialBrandId` effect in story book. 
wondering whether we have the initialBrandId changed in real scenario or not? to consider if it's necessary